### PR TITLE
feat: BYOB verification support

### PR DESCRIPTION
**Pull request by laurentsimon**
closes https://github.com/slsa-framework/slsa-verifier/issues/601

CLI e2e tests will be tacked as part of https://github.com/slsa-framework/slsa-verifier/issues/603

### CODE REVIEW

**Comment by laurentsimon** [comment]
@ianlewis do you know what's happening with https://github.com/slsa-framework/slsa-verifier/actions/runs/5019866408/jobs/9000799655?pr=604?

File: verifiers/utils/structures.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package utils
+
+func MergeMaps[K comparable, V any](m1, m2 map[K]V) map[K]V {
+	m := make(map[K]V, 0)
```
[the remark this answers is not in the document]
Comment (laurentsimon) [reply verifiers/utils/structures.go:4]: +1, I forgot this. Good catch!

**Comment by laurentsimon** [comment]
@ianlewis I made the change to use 1.18 go, but still encountering the same problem. Any ideas?

**Comment by ianlewis** [comment]
> @ianlewis I made the change to use 1.18 go, but still encountering the same problem. Any ideas?

I think they disable checks that don't support generecs yet when you set that, but they missed `gocritic` since I think it's not enabled by default. Let's just update golangci-lint since I think this issue is already fixed.

**Comment by laurentsimon** [comment]
> > @ianlewis I made the change to use 1.18 go, but still encountering the same problem. Any ideas?
> 
> I think they disable checks that don't support generics yet when you set that, but they missed `gocritic` since I think it's not enabled by default. Let's just update golangci-lint since I think this issue is already fixed.

Thanks!

**Comment by laurentsimon** [comment]
> > > @ianlewis I made the change to use 1.18 go, but still encountering the same problem. Any ideas?
> > 
> > 
> > I think they disable checks that don't support generics yet when you set that, but they missed `gocritic` since I think it's not enabled by default. Let's just update golangci-lint since I think this issue is already fixed.
> 
> Thanks!

It's now failing with `non-wrapping format verb for fmt.Errorf. Use `%w` to format errors (errorlint)`. Shall I add a `//nolint:errorlint`? I don't see what's wrong with the code, ie it's already using `%w`

**Comment by ianlewis** [comment]
> > > > @ianlewis I made the change to use 1.18 go, but still encountering the same problem. Any ideas?
> > > 
> > > 
> > > I think they disable checks that don't support generics yet when you set that, but they missed `gocritic` since I think it's not enabled by default. Let's just update golangci-lint since I think this issue is already fixed.
> > 
> > 
> > Thanks!
> 
> It's now failing with `non-wrapping format verb for fmt.Errorf. Use `%w` to format errors (errorlint)`. Shall I add a `//nolint:errorlint`? I don't see what's wrong with the code, ie it's already using `%w`

Yeah, It's complaining because Go 1.20 allows you to wrap multiple errors (aside: which obviates the need to have the wrappableError in the generators). I added a config to ignore it for now.

**Comment by laurentsimon** [comment]
> > It's now failing with `non-wrapping format verb for fmt.Errorf. Use `%w` to format errors (errorlint)`. Shall I add a `//nolint:errorlint`? I don't see what's wrong with the code, ie it's already using `%w`
> 
> Yeah, It's complaining because Go 1.20 allows you to wrap multiple errors (aside: which obviates the need to have the wrappableError in the generators). I guess the linter doesn't know we're still using Go 1.18. I added a config to ignore it for now.

Thanks. can't wait to use 1.20!

**Pull request merged**